lib/chef-dk/command/verify.rb

```diff
@@ -141,6 +147,35 @@ def components
           # ------------
           failures = []
 
+          # fail the verify if we have more than one version of chef-provisioning or any of its drivers.
```

Comment:
We've discussed doing this for chef-dk as a whole; one thing that hasn't come up is that this means `chef verify` will fail for someone's ChefDK setup where they have updated or installed gems, and whether that's okay.